config/tests/samples/create/harness.go

```diff
@@ -305,17 +311,40 @@ func NewHarness(ctx context.Context, t *testing.T) *Harness {
 		kccConfig.HTTPClient = &http.Client{Transport: t}
 	}
 
-	// Intercept (and log) TF requests
-	transport_tpg.DefaultHTTPClientTransformer = func(ctx context.Context, inner *http.Client) *http.Client {
-		ret := inner
-		if t := ctx.Value(httpRoundTripperKey); t != nil {
-			ret = &http.Client{Transport: t.(http.RoundTripper)}
+	if targetGCP := os.Getenv("E2E_GCP_TARGET"); targetGCP == "vcr" {
```

Comment:
Non-blocking refactoring idea:
Can we do one `targetGCP := os.Getenv("E2E_GCP_TARGET")` in the beginning of the function, and then do switch case/if else for different conditions in line233-247, line314 and one, etc?

Repetitively retrieving the same environment seems to reduce the readability.

Reply:
this will be done in b/332719149 as well.